src/providers/FileSystemProvider/TextSearchProvider.ts

```diff
@@ -104,13 +104,43 @@ export class TextSearchProvider implements vscode.TextSearchProvider {
     } else {
       const sysStr = params.has("system") && params.get("system").length ? params.get("system") : "0";
       const genStr = params.has("generated") && params.get("generated").length ? params.get("generated") : "0";
+
+      let uri = options.folder;
+
+      if (!new URLSearchParams(uri.query).has("filter")) {
```

Comment:
Can we allow this when the filter is present and just append our new filter to the existing one? I'm not sure if that could lead to a malformed filter but it's worth considering.

Reply:
I had wondered about doing that, but if the isfs spec's filter already lists what to include isn't it the case that adding more includes can never result in less server-side searching, and may unexpectedly cause more? Adding excludes might be OK I suppose.

Reply:
Yes, I suppose it could lead to more searching if you put something in the includes that isn't in the folder to begin with, like if your folder has `type=cls` but you put `*.mac` in the includes. That would be really dumb but there's nothing stopping you from doing that. Excludes should be fine.